pkg/client/keycloak/config.go

```diff
@@ -130,6 +129,14 @@ func (kc *KeycloakConfig) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&kc.SsoBaseUrl, "redhat-sso-base-url", kc.SsoBaseUrl, "The base URL of the mas-sso, integration by default")
 	fs.StringVar(&kc.SSOSpecialManagementOrgID, "sso-special-management-org-id", SSO_SPEICAL_MGMT_ORG_ID_STAGE, "The Special Management Organization ID used for creating internal Service accounts")
 	fs.StringVar(&kc.ServiceAccounttLimitCheckSkipOrgIdListFile, "service-account-limits-check-skip-org-id-list-file", kc.ServiceAccounttLimitCheckSkipOrgIdListFile, "File containing a list of Org IDs for which service account limits check will be skipped")
+	fs.StringVar(&kc.SelectSSOProvider, "sso-provider-type", kc.SelectSSOProvider, "Option to choose between sso providers i.e, mas-sso or redhat_sso, mas-sso by default")
+}
+
+func (kc *KeycloakConfig) Validate(env *environments.Env) error {
+	if kc.SelectSSOProvider != REDHAT_SSO && kc.SelectSSOProvider != MAS_SSO {
+		return fmt.Errorf("Invalid sso provider selcted must be `mas-sso` or `redhat_sso`")
+	}
+	return nil
```

Comment:
As in my previous comment: if we keep `selectSSOProvider` as object, we can add a `Validate` method to that object. 
That way if in the future the list of supported provider should change, the client code won't need any update.

Reply:
I think this also applies to my comment above. If I am wrong could you clarify please? https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1007#discussion_r873779876